framework/src/modules/dpos_v2/commands/pom.ts

```diff
@@ -33,47 +33,54 @@ import {
 	TokenMethod,
 	TokenIDDPoS,
 	ValidatorsMethod,
+	ModuleConfig,
 } from '../types';
-import { getPunishmentPeriod } from '../utils';
+import { getDelegateWeight, getPunishmentPeriod } from '../utils';
 import { ValidationError } from '../../../errors';
 import { areDistinctHeadersContradicting } from '../../../engine/bft/utils';
 import { DelegateStore } from '../stores/delegate';
+import { DelegatePunishedEvent } from '../events/delegate_punished';
+import { DelegateBannedEvent } from '../events/delegate_banned';
+import { EligibleDelegatesStore } from '../stores/eligible_delegates';
 
-export class ReportDelegateMisbehaviorCommand extends BaseCommand {
+export class ReportMisbehaviorCommand extends BaseCommand {
 	public schema = pomCommandParamsSchema;
 	private _tokenMethod!: TokenMethod;
 	private _validatorsMethod!: ValidatorsMethod;
 	private _tokenIDDPoS!: TokenIDDPoS;
+	private _config!: ModuleConfig;
 
 	public addDependencies(args: PomCommandDependencies) {
 		this._tokenMethod = args.tokenMethod;
 		this._validatorsMethod = args.validatorsMethod;
 	}
 
-	public init(args: { tokenIDDPoS: TokenIDDPoS }) {
+	public init(args: { tokenIDDPoS: TokenIDDPoS; config: ModuleConfig }) {
 		this._tokenIDDPoS = args.tokenIDDPoS;
+		this._config = args.config;
 	}
 
 	// eslint-disable-next-line @typescript-eslint/require-await
 	public async verify(
 		context: CommandVerifyContext<PomTransactionParams>,
 	): Promise<VerificationResult> {
+		const { params } = context;
+
 		try {
-			validator.validate(this.schema, context.params);
+			validator.validate(this.schema, params);
 		} catch (err) {
 			return {
 				status: VerifyStatus.FAIL,
 				error: err as Error,
 			};
 		}
 
-		const header1 = BlockHeader.fromBytes(context.params.header1);
-		header1.validate();
+		const header1 = BlockHeader.fromBytes(params.header1);
+		const header2 = BlockHeader.fromBytes(params.header2);
 
-		const header2 = BlockHeader.fromBytes(context.params.header2);
+		header1.validate();
 		header2.validate();
 
```

Comment:
i think still some of the verification is missing here which exist in the execute